modules/components/social/org.wso2.carbon.social.sql/src/main/java/org/wso2/carbon/social/sql/SocialUtil.java

```diff
@@ -44,6 +47,20 @@ public static String getTenantDomain() {
 
 	}
 
+	public static String getTenantDomainFromTenantId(int tenantId) throws SocialActivityException{
+		String tenantDomainName = null;
+		try {
+			UserRealmService realmService = (UserRealmService) PrivilegedCarbonContext.getThreadLocalCarbonContext().
+					getOSGiService(UserRealmService.class);
+			org.wso2.carbon.user.api.TenantManager tenantManager = realmService.getTenantManager();
+			tenantDomainName = tenantManager.getTenant(tenantId).getDomain();
```

Comment:
realmService and tenantManager won't be null. If so it must rise before this point. If tenantManager.getTenant(tenantId) is null it will throw UserStoreException